classes/fullstack/project1/server/src/services/mongodb/build-service.ts

```diff
@@ -0,0 +1,121 @@
+import { BuildCountWeekly, IBuild } from 'types'
+import { BuildModel } from './models'
+import { BUILD_AGREGATION_START_OF_WEEK } from '../../config'
+import { formatISO } from 'date-fns'
+
+export const buildService = {
+  getBuildsPaginated: async ({
+    page = 1,
+    limit = Number.MAX_SAFE_INTEGER,
+  }: Pagination = {}): Promise<IBuildWithId[]> => {
+    try {
+      const raw = await BuildModel.find()
+        .skip((page - 1) * limit)
+        .limit(limit)
+
+      return raw.map((build) => {
+        const {
+          _id,
+          endTime,
+          startTime,
+          errorsNumber,
+          warningsNumber,
+          ...rest
+        } = build.toObject()
+
+        return {
+          _id: _id.toString(),
+          startTime: formatISO(startTime),
+          endTime: formatISO(endTime),
+          // TODO: Consider correcting IBuild.errorsNumber type to number
+          errorsNumber: errorsNumber.toString(),
+          // TODO: Consider correcting IBuild.warningsNumber type to number
+          warningsNumber: warningsNumber.toString(),
+          ...rest,
+        }
+      })
+    } catch (error) {
+      console.error(error)
+      return []
+    }
+  },
+  getBuildById: async (buildId: string): Promise<IBuildWithId | null> => {
+    try {
+      const build = await BuildModel.findById(buildId)
+
+      if (build) {
```

Comment:
same here...no need

Reply:
Corrected to return raw data without any processing, adjusted jest tests